Add guard for named tensors in the JIT

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25345 Add guards for using named tensor with serialization and multiprocessing
* **#25344 Add guard for named tensors in the JIT**

Test Plan
- [namedtensor ci]

Differential Revision: [D17101487](https://our.internmc.facebook.com/intern/diff/D17101487)

zou3519 committed:
Add guard for named tensors in the JIT

Test Plan
- [namedtensor ci]

nairbv reviewed (approved):
seems reasonable to me but might want to wait for @eellison to approve.

Will jit eventually support named tensors?

eellison reviewed (approved):
LGTM. 

Note that this only checks border between python and JIT, so if someone uses Named Tensors in the middle of a script method this won't get hit. But there isn't really a better option, and this is also what we did for sparse tensors.

zou3519 committed:
Update on "Add guard for named tensors in the JIT"


Test Plan
- [namedtensor ci]

Differential Revision: [D17101487](https://our.internmc.facebook.com/intern/diff/D17101487)

eellison commented on test/test_namedtensor.py:
```diff
@@ -799,6 +799,15 @@ def test_as_strided(self):
+            return x + 1
+
+        named_tensor = torch.randn(2, 3, names=('N', 'C'))
+        with self.assertRaisesRegex(RuntimeError, 'NYI'):
```
Yea returning a named tensor into python from script.

zou3519 replied:
Cool, I assumed that and added a test before I landed.
